File: TripsServer/Trips/Entities/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Trips.Entities
{
    public class Vehicle
    {
        public int Id { get; set; }
        public User Owner { get; set; }
        public string Name { get; set; }
        public string OfficialName { get; set; }
        public string Description { get; set; }
        public string LicenseNumber { get; set; }
        public int? YearOfManufacture { get; set; }
        public PlaceAccessibility? AcceptableAccessibility { get; set; }
        public Gallery Gallery { get; set; }
        public Picture TitlePicture { get; set; }

        public User AddedBy { get; set; }
        public DateTime? AddedDate { get; set; }
        public User ChangedBy { get; set; }
        public DateTime? ChangedDate { get; set; }

        public ICollection<VehiclesToTrips> Trips { get; set; }
    }
}

File: trips-client/src/app/models/vehicle.ts
import { UserHeader } from './user-header';
import { PlaceAccessibility } from './place';
import { Gallery } from './gallery';
import { Picture } from './picture';

export class Vehicle {
    id: number;
    owner: UserHeader;
    name: string;
    officialName: string;
    description: string;
    licenseNumber: string;
    yearOfManufacture: number;
    acceptableAccessibility: PlaceAccessibility;
    gallery: Gallery;
    titlePicture: Picture;
    addedBy: UserHeader;
    addedDate: Date;
    changedBy: UserHeader;
    changedDate: Date;
}